packages/firestore/src/core/component_provider.ts

```diff
@@ -350,23 +349,25 @@ export class MultiTabOfflineComponentProvider extends IndexedDbOfflineComponentP
         this.onlineComponentProvider.syncEngine,
         isPrimary
       );
-      if (this.gcScheduler) {
-        if (isPrimary && !this.gcScheduler.started) {
-          this.gcScheduler.start();
-        } else if (!isPrimary) {
-          this.gcScheduler.stop();
-        }
-      }
-      if (this.indexBackfillerScheduler) {
-        if (isPrimary && !this.indexBackfillerScheduler.started) {
-          this.indexBackfillerScheduler.start();
-        } else if (!isPrimary) {
-          this.indexBackfillerScheduler.stop();
-        }
-      }
+      this.startOrStopScheduler(this.gcScheduler, isPrimary);
+      this.startOrStopScheduler(this.indexBackfillerScheduler, isPrimary);
     });
   }
 
+  private startOrStopScheduler(
+    scheduler: Scheduler | null,
+    isPrimary: boolean
```

Comment:
Perhaps a comment as to what `isPrimary` represents.

Reply:
Done.